src/synth/fluid_synth.c

```diff
@@ -3652,15 +3652,24 @@ fx[ ((k * fluid_synth_count_effects_channels() + j) * 2 + 1) % nfx ]  = right_bu
  *
  * @param synth FluidSynth instance
  * @param len Count of audio frames to synthesize and store in every single buffer provided by \p out and \p fx.
- * @param nfx Count of arrays in \c fx. Must be a multiple of 2 (because of stereo)
- * and in the range <code>0 <= nfx/2 <= (fluid_synth_count_effects_channels() * fluid_synth_count_effects_groups())</code>.
+ *  Zero value is permitted, the function does nothing and return FLUID_OK.
+ * @param nfx Count of arrays in \c fx. Must be a multiple of 2 (because of stereo).
+ * and in the range <code>0 <= nfx/2 <= (fluid_synth_count_effects_channels() * fluid_synth_count_effects_groups()).
+  Note that zero value is valid and allows to skip mixing effects in all fx output buffers.</code>.
  * @param fx Array of buffers to store effects audio to. Buffers may
-alias with buffers of \c out. NULL buffers are permitted and will cause to skip mixing any audio into that buffer.
+alias with buffers of \c out. Individual NULL buffers are permitted and will cause to skip mixing any audio into that buffer.
  * @param nout Count of arrays in \c out. Must be a multiple of 2
-(because of stereo) and in the range <code>0 <= nout/2 <= fluid_synth_count_audio_channels()</code>.
+(because of stereo) and in the range <code>0 <= nout/2 <= fluid_synth_count_audio_channels().
+ Note that zero value is valid and allows to skip mixing dry audio in all out output buffers.</code>.
  * @param out Array of buffers to store (dry) audio to. Buffers may
-alias with buffers of \c fx. NULL buffers are permitted and will cause to skip mixing any audio into that buffer.
- * @return #FLUID_OK on success, #FLUID_FAILED otherwise.
+alias with buffers of \c fx. Individual NULL buffers are permitted and will cause to skip mixing any audio into that buffer.
+ * @return #FLUID_OK on success,
+ * #FLUID_FAILED otherwise,
+ *  - fx NULL while nfx > 0, or out NULL while nout > 0.
+ *  - nfx or nout not multiple of 2.
+ *  - len < 0.
+ *  - both nfx and nout set to 0 while len > 0
+ *  - nfx or nout greater than respective internal mixer buffer count.
```

Comment:
```suggestion
 *  - <code>fx == NULL</code> while <code>nfx > 0</code>, or <code>out == NULL</code> while <code>nout > 0</code>.
 *  - \c nfx or \c nout not multiple of 2.
 *  - <code>len < 0</code>.
 *  - \c nfx or \c nout exceed the range explained above.
```

Reply:
Thanks. I assume that these xml-like notation will be read by a tool that produce formatted documentation in the Wiki, isn't it ?

Reply:
The tool is called doxygen and generates the [API doc](http://fluidsynth.org/api). Not to be confused with the [wiki](https://github.com/FluidSynth/fluidsynth/wiki).

Reply:
>The tool is called doxygen and generates the API doc and supports embedding HTML.

Ok, thanks.

> Not to be confused with the wiki.

Yes. In fact the wiki is only manually "generated".